pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go

```diff
@@ -51,7 +51,7 @@ var cmpOpts = []cmp.Option{
 }
 
 var (
-	topologySpreadFunc = frameworkruntime.FactoryAdapter(feature.Features{}, New)
+	topologySpreadFunc = frameworkruntime.FactoryAdapter(context.Background(), feature.Features{}, New)
```

Comment:
When migrating podtopologyspread to contextual logging, we should get rid of the background context.

Reply:
Sorry for the delay due to K8s code freeze and some other things.

Of course, but in that case, we need to get ctx from the caller (from test) instead.
I talked to pohly about how to deliver ctx in this situation.

>Those must be converted to functions that accept a context. Then when they are called(for example from a test), the caller will be responsible for creating that context.

for example:
before：
```
topologySpreadFunc = frameworkruntime.FactoryAdapter(feature.Features{}, New)
```
after:
```
func topologySpreadFunc(ctx context.Contex) PluginFactory {
    return frameworkruntime.FactoryAdapter(ctx, feature.Features{}, New)
}
```
in tests function, we need to call `topologySpreadFunc(ctx)` to pass ctx.

Reply:
The latest commit https://github.com/kubernetes/kubernetes/pull/116884/commits/8d94ab5735e6496217b7dabf8b7ed1c054ad7062 to implement the above. Please review it again. @alculquicondor